common/app/views/fragments/omnitureScript.scala.html

```diff
@@ -0,0 +1,28 @@
+@(item: Option[model.MetaData])
+
+@import conf.Switches.InlineOmnitureScript
+@import conf.Static
+@import conf.Configuration
+@import views.support.OmnitureAnalyticsAccount
+
+@item.map{ item =>
+    <script>
+        @* must always be set before the Omniture file is parsed *@
+        window.s_account = '@OmnitureAnalyticsAccount(item)';
+    </script>
+}
+
+@if(InlineOmnitureScript.isSwitchedOn) {
+    <script>
+        try {
+            @Html(Static.js.omnitureJs)
+        } catch (e) {
+            (new Image()).src = '@{Configuration.debug.beaconUrl}/count/omniture-library-error.gif';
+        }
+    </script>
+} else {
+    @* hardly ever changes, so extremely cacheable *@
+    @* unfortunately we cannot put 'defer' on it, hence the ability
+    to switch between this and an inline version for testing*@
+    <script id="omniture" src="@Static("javascripts/vendor/omniture.js")"></script>
```

Comment:
Its not an A/B test, it is switched 100% to one or the other.

We will probably only try the blocking script for a short time (if at all).

Grant

On 10 September 2015 at 16:40, Oliver Joseph Ash <notifications@github.com>
wrote:

> In common/app/views/fragments/omnitureScript.scala.html
> <https://github.com/guardian/frontend/pull/10452#discussion_r39175224>:
>
> > +    </script>
> > +}
> > +
> > +@if(InlineOmnitureScript.isSwitchedOn) {
> > +    <script>
> > +        try {
> > +            @Html(Static.js.omnitureJs)
> > +        } catch (e) {
> > +            (new Image()).src = '@{Configuration.debug.beaconUrl}/count/omniture-library-error.gif';
> > +        }
> > +    </script>
> > +} else {
> > +    @* hardly ever changes, so extremely cacheable *@
> > +    @* unfortunately we cannot put 'defer' on it, hence the ability
> > +    to switch between this and an inline version for testing*@
> > +    <script id="omniture" src="@Static("javascripts/vendor/omniture.js")"></script>
>
> If I am seeing this correctly, for users not in the test, they will have a
> parser blocking script tag to download. That's not good [image:
> :crying_cat_face:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/10452/files#r39175224>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396